consul/src/main/scala/io/buoyant/consul/v1/ConsulApi.scala

```diff
@@ -135,10 +135,10 @@ class HealthApi(
           checks.reduceOption(HealthStatus.worstCase)
         )
       }
-      val nodes = statuses match {
-        case None => result
-        case Some(x) if x contains HealthStatus.Any => result
-        case Some(x) => result.filter(x contains _.Status.getOrElse(HealthStatus.Passing))
+      val nodes = if (statuses == Set(HealthStatus.Passing)) {
+        result
+      } else {
+        result.filter(statuses contains _.Status.getOrElse(HealthStatus.Passing))
```

Comment:
Our preferred code style tends to discourage the use of whitespace-delimited function calls. Prefer:
```scala
result.filter(statuses.contains(_.Status.getOrElse(HealthStatus.Passing)))
```